nilearn/interfaces/fmriprep/load_confounds_utils.py

```diff
@@ -89,15 +89,22 @@ def _get_file_name(nii_file):
     """Construct the raw confound file name from processed functional data."""
     if isinstance(nii_file, list):  # catch gifti
         nii_file = nii_file[0]
-    suffix = "_space-" + nii_file.split("space-")[1]
+    entities = parse_bids_filename(nii_file)
+    subject_label = f"sub-{entities['sub']}"
+    if "session" in entities:
```

Comment:
Still wrapping my head around how nilearn deals with BIDS things but isn't that supposed to be 

```python
if "ses" in entities:
```

Reply:
Yes, and the test is missing this line. Will fix it